SU2_CFD/src/solvers/CScalarSolver.cpp

```diff
@@ -0,0 +1,799 @@
+       introduction of the GCL term above, the remainder of the source residual
+       due to the time discretization has a new form.---*/
+
+      if (Conservative) {
+        /*--- If this is the SST model, we need to multiply by the density
+         in order to get the conservative variables ---*/
+        if (incompressible) {
+          /*--- This is temporary and only valid for constant-density problems:
+          density could also be temperature dependent, but as it is not a part
+          of the solution vector it's neither stored for previous time steps
+          nor updated with the solution at the end of each iteration. */
+          Density_nM1 = flowNodes->GetDensity(iPoint);
+          Density_n = flowNodes->GetDensity(iPoint);
+          Density_nP1 = flowNodes->GetDensity(iPoint);
+        } else {
+          Density_nM1 = flowNodes->GetSolution_time_n1(iPoint)[0];
+          Density_n = flowNodes->GetSolution_time_n(iPoint, 0);
+          Density_nP1 = flowNodes->GetSolution(iPoint, 0);
+        }
+      }
+
+      for (iVar = 0; iVar < nVar; iVar++) {
+        if (first_order)
+          LinSysRes(iPoint, iVar) +=
+              (Density_nP1 * U_time_nP1[iVar] - Density_n * U_time_n[iVar]) * (Volume_nP1 / TimeStep);
+        if (second_order)
+          LinSysRes(iPoint, iVar) +=
+              (Density_nP1 * U_time_nP1[iVar] - Density_n * U_time_n[iVar]) * (3.0 * Volume_nP1 / (2.0 * TimeStep)) +
+              (Density_nM1 * U_time_nM1[iVar] - Density_n * U_time_n[iVar]) * (Volume_nM1 / (2.0 * TimeStep));
+      }
+
+      /*--- Compute the Jacobian contribution due to the dual time source term. ---*/
+      if (implicit) {
+        if (first_order) Jacobian.AddVal2Diag(iPoint, Volume_nP1 / TimeStep);
+        if (second_order) Jacobian.AddVal2Diag(iPoint, (Volume_nP1 * 3.0) / (2.0 * TimeStep));
+      }
+    }
+    END_SU2_OMP_FOR
+
+    AD::EndNoSharedReading();
+
+  }  // end dynamic grid
+}
+
+void CScalarSolver::LoadRestart(CGeometry** geometry, CSolver*** solver, CConfig* config, int val_iter,
+                                bool val_update_geo) {
+  /*--- Restart the solution from file information ---*/
+
+  unsigned short iVar, iMesh;
+  unsigned long iPoint, index, iChildren, Point_Fine;
+  su2double Area_Children, Area_Parent;
+  const su2double* Solution_Fine = nullptr;
+
+  string restart_filename = config->GetFilename(config->GetSolution_FileName(), "", val_iter);
+
+  /*--- To make this routine safe to call in parallel most of it can only be executed by one thread. ---*/
+  SU2_OMP_MASTER {
+    /*--- Read the restart data from either an ASCII or binary SU2 file. ---*/
+
+    if (config->GetRead_Binary_Restart()) {
+      Read_SU2_Restart_Binary(geometry[MESH_0], config, restart_filename);
+    } else {
+      Read_SU2_Restart_ASCII(geometry[MESH_0], config, restart_filename);
+    }
+
+    /*--- Skip flow variables ---*/
+
+    unsigned short skipVars = nDim + solver[MESH_0][FLOW_SOL]->GetnVar();
```

Comment:
`CScalarSolver::LoadRestart` will work like this only in a 'choose one' fashion. For Turb and Species Transport There needs to be a clear hierarchy (which is given by the ouptut of course). Although not strictly necessary (I think), keeping the order of how the linear systems are solved might be good for OCD-patients. 
I personally would keep this method right here because the majority stays the same. idk have a `skipVars = DetermineSkpVars(config)` method